src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/mapper/induction/InductionEntityMapper.kt

```diff
@@ -78,31 +78,61 @@ abstract class InductionEntityMapper {
   @Mapping(target = "workOnRelease", expression = "java( updateWorkOnRelease(entity, dto) )")
   abstract fun updateEntityFromDto(@MappingTarget entity: InductionEntity, dto: UpdateInductionDto)
 
-  fun updateFutureWorkInterests(entity: InductionEntity, dto: UpdateInductionDto): FutureWorkInterestsEntity? =
-    futureWorkInterestsEntityMapper.updateEntityFromDto(entity.futureWorkInterests, dto.futureWorkInterests)
-      .let { entity.futureWorkInterests }
-
-  fun updateInPrisonInterests(entity: InductionEntity, dto: UpdateInductionDto): InPrisonInterestsEntity? =
-    inPrisonInterestsEntityMapper.updateEntityFromDto(entity.inPrisonInterests, dto.inPrisonInterests)
-      .let { entity.inPrisonInterests }
-
-  fun updatePersonalSkillsAndInterests(entity: InductionEntity, dto: UpdateInductionDto): PersonalSkillsAndInterestsEntity? =
-    personalSkillsAndInterestsEntityMapper.updateEntityFromDto(entity.personalSkillsAndInterests, dto.personalSkillsAndInterests)
-      .let { entity.personalSkillsAndInterests }
-
-  fun updatePreviousQualifications(entity: InductionEntity, dto: UpdateInductionDto): PreviousQualificationsEntity? =
-    previousQualificationsEntityMapper.updateEntityFromDto(entity.previousQualifications, dto.previousQualifications)
-      .let { entity.previousQualifications }
-
-  fun updatePreviousTraining(entity: InductionEntity, dto: UpdateInductionDto): PreviousTrainingEntity? =
-    previousTrainingEntityMapper.updateEntityFromDto(entity.previousTraining, dto.previousTraining)
-      .let { entity.previousTraining }
-
-  fun updatePreviousWorkExperiences(entity: InductionEntity, dto: UpdateInductionDto): PreviousWorkExperiencesEntity? =
-    previousWorkExperiencesEntityMapper.updateEntityFromDto(entity.previousWorkExperiences, dto.previousWorkExperiences)
-      .let { entity.previousWorkExperiences }
+  fun updateFutureWorkInterests(entity: InductionEntity, dto: UpdateInductionDto): FutureWorkInterestsEntity? {
+    return if (entity.futureWorkInterests == null) {
```

Comment:
If the original entity is null, then we can't call the existing `updateEntityFromDto` methods, since they use the `@MappingTarget annotation`, which relies on a non null target object.